### ISSUE TITLE
ThrowingExceptionsWithoutMessageOrCause false positive

### ISSUE THREAD

**Issue by o-shevchenko** [detekt/detekt#7448]
## Expected Behavior
```kotlin
loader.errors().firstOrNull()?.exception()?.let { throw it }
```
The following code should work fine and don't violate ThrowingExceptionsWithoutMessageOrCause since we already have a message for an exception

## Observed Behavior

```kotlin
loader.errors().firstOrNull()?.exception()?.let { throw it }
```
Throws ThrowingExceptionsWithoutMessageOrCause
Even the following code violates the rule
```kotlin
loader.errors().firstOrNull()?.exception()?.let { throw DataAccessException(it.message, it.cause) }
```
For some reason it reacts on method invocation `.exception()`

## Context

I'm using jooq Loader API
https://www.jooq.org/doc/latest/manual/sql-execution/importing/importing-api/

## Your Environment

* Version of detekt used: 1.23.6
* Version of Gradle used (if applicable): 8.6
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: MacOS
* Link to your project (if it's a public repository):

**Comment by o-shevchenko** [comment]
The problem still exists

**Comment by o-shevchenko** [comment]
Can we remove `stale` label to prevent issue closing?
Thanks

**Comment by BraisGabin** [comment]
I think that this Rule is wrong. We shouldn't fail when an exception without message nor cause is *throwed*. We should fail when an exception without message nor cause is *created*.

**Comment by BraisGabin** [comment]
Ok, this rule doesn't care about `throw`. It only cares about the creation of the rule (the name is a bit missleading). And I found the issue. It thinks that `.exception()` is a call to the `Exception` constructor and for that reason it raises an issue. I created a PR to fix it.
